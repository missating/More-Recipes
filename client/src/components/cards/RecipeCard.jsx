import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


const RecipeCard = props => (
  <div className="flex">

    <div className="card">
      <Link to={`/recipes/view/${props.id}`} className="hover">
        <img
          src={props.recipeImage}
          alt={props.recipeImage}
          className="image-width"
        />
      </Link>
      <div className="container p-3 recipedetails-column">
        <h2>{props.name}</h2>
        <p><small>{props.description}</small></p>
        <div className="center-button">
          <Link
            className="btn btn-secondary"
            to={`/recipes/view/${props.id}`}
          >
            VIEW
          </Link>
          {
            props.button &&
            (
              <button
                id="removeButton"
                className="btn btn-secondary"
                onClick={() => props.onButtonClick(props.id)}
              >
                Remove
              </button>
            )

          }
        </div>
      </div>
    </div>
  </div>
);


RecipeCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  recipeImage: PropTypes.string.isRequired,
  onButtonClick: PropTypes.func,
  button: PropTypes.bool
};

RecipeCard.defaultProps = {
  onButtonClick: () => null,
  button: false
};

export default RecipeCard;
