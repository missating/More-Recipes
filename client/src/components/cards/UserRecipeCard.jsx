import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { confirmAlert } from 'react-confirm-alert';
import 'react-confirm-alert/src/react-confirm-alert.css';
import { connect } from 'react-redux';

// actions
import deleteRecipe from '../../actions/deleteRecipe';
/**
 * Creates UserRecipeCard component
 * @class UserRecipeCard
 * @extends {React.Component}
 */
export class UserRecipeCard extends React.Component {
  /**
   * Creates an instance of UserRecipeCard.
   * @param {any} props
   * @memberof UserRecipeCard
   */
  constructor(props) {
    super(props);
    this.onDeleteRecipe = this.onDeleteRecipe.bind(this);
  }
  /**
   * Deletes a recipe after confirming
   * @method onDeleRecipe
   * @param {any} event
   * @return {void}
   * @memberof UserRecipeCard
   */
  onDeleteRecipe() {
    confirmAlert({
      title: '',
      message: 'Are you sure you want to delete this recipe ?',
      confirmLabel: 'Yes, delete!',
      cancelLabel: 'Cancel',
      onConfirm: () =>
        this.props.deleteRecipe(this.props.id),
    });
  }
  /**
   * Renders react component
   * @method render
   * @returns {void}
   * @memberof  UserRecipeCard
   */
  render() {
    return (
      <div className="flex">

        <div className="card">
          <Link to={`/recipes/view/${this.props.id}`} className="hover">
            <img
              src={this.props.recipeImage}
              alt={this.props.recipeImage}
              className="image-width"
            />
          </Link>
          <div className="container p-3 recipedetails-column">
            <h2>{this.props.name}</h2>
            <p><small>{this.props.description}</small></p>
            <div className="container">
              <Link
                className="btn btn-secondary recipe-button"
                to={`/recipes/view/${this.props.id}`}
              >
                VIEW
              </Link>

              <Link
                className="btn btn-secondary recipe-button"
                to={`/recipe/edit/${this.props.id}`}
              >
                EDIT
              </Link>

              <button
                id="deleteButton"
                className="btn"
                onClick={this.onDeleteRecipe}
              >
                <span>
                  <i className="far fa-trash-alt" />
                </span>
              </button>

            </div>
          </div>
        </div>

      </div>
    );
  }
}

UserRecipeCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  recipeImage: PropTypes.string.isRequired,
  deleteRecipe: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => ({
  deleteRecipe: (recipeId) => {
    dispatch(deleteRecipe(recipeId));
  }
});

const ConnectedComponent =
  connect(null, mapDispatchToProps)(UserRecipeCard);
export { ConnectedComponent as ConnectedUserRecipeCard };

