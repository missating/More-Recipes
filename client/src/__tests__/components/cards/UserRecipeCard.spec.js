import React from 'react';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import { UserRecipeCard } from '../../../components/cards/UserRecipeCard';

jest.mock('react-confirm-alert', () => ({
  confirmAlert: jest.fn()
}));

describe('User recipe card component', () => {
  const props = {
    id: 1,
    name: 'test',
    description: 'mix it',
    recipeImage: 'image_url'
  };

  const deleteRecipe = jest.fn(() => Promise.resolve());

  it('Should render', () => {
    const wrapper = shallow(<UserRecipeCard
      {...props}
      deleteRecipe={deleteRecipe}
    />);
    expect(wrapper.find('div').length).toEqual(4);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Should delete recipe when button is clicked', () => {
    const spy = sinon.spy(UserRecipeCard.prototype, 'onDeleteRecipe');
    const wrapper = shallow(<UserRecipeCard
      {...props}
      deleteRecipe={deleteRecipe}
    />);
    wrapper.find('#deleteButton').simulate('click');
    expect(spy.called).toBeTruthy();
  });
});

