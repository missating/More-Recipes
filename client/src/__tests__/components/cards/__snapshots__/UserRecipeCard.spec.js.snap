// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User recipe card component Should render 1`] = `
<div
  className="flex"
>
  <div
    className="card"
  >
    <Link
      className="hover"
      replace={false}
      to="/recipes/view/1"
    >
      <img
        alt="image_url"
        className="image-width"
        src="image_url"
      />
    </Link>
    <div
      className="container p-3 recipedetails-column"
    >
      <h2>
        test
      </h2>
      <p>
        <small>
          mix it
        </small>
      </p>
      <div
        className="container"
      >
        <Link
          className="btn btn-secondary recipe-button"
          replace={false}
          to="/recipes/view/1"
        >
          VIEW
        </Link>
        <Link
          className="btn btn-secondary recipe-button"
          replace={false}
          to="/recipe/edit/1"
        >
          EDIT
        </Link>
        <button
          className="btn"
          id="deleteButton"
          onClick={[Function]}
        >
          <span>
            <i
              className="far fa-trash-alt"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
